### PR TITLE
chore(linter): Add cosmos gosec CI job

### PR DESCRIPTION
**Pull request by sesheffield**
## Description
This PR integrates gosec with the golangci-lint tool and configures it to exclude generated files. The linter runs on modified files, consistent with the behavior of other Go linters.

Future iterations will include custom linters to run specific cosmos/gosec rules.
https://app.shortcut.com/kava-labs/story/13970/add-custom-gosec-linter

## Checklist
 - [ ] Changelog has been updated as necessary.

### CODE REVIEW

File: .github/workflows/ci-commit.yml
Comment (pirtleshell) [inline .github/workflows/ci-commit.yml:9]: do we need this in `ci-pr.yml` (or `ci-default.yml`) to ensure it runs on & blocks merge of PRs?

asking b/c i don't know :)

Comment (sesheffield) [reply .github/workflows/ci-commit.yml:9]: I'll add it to `ci-default.yml` since that workflow is being called in `ci-pr.yml` anyways

**Review by nddeluca — CHANGES_REQUESTED** [review]
The gosec addition looks good, but this needs some cleanup before merge.

The .golangci-yml should end in a new line, as well as the ci-default.yml file (though this should not have been committed?).

In addition, the history is messy -- can we rebase and clean it up?

**Review by nddeluca — APPROVED** [review]
Nice work, I like the re-ordering of the linter-setting section to be alphabetical 👍

File: .golangci.yml
Comment (nddeluca) [inline .golangci.yml:113]: Did you manually re-order these or was this automatic in an editor/linter/etc?

Comment (sesheffield) [reply .golangci.yml:113]: this was manual, but there are tools to automatically sort lines and structures

Comment (sesheffield) [reply .golangci.yml:113]: https://plugins.jetbrains.com/plugin/14407-smartsort

**Pull request merged**
